substrate/frame/election-provider-multi-block/src/lib.rs

```diff
@@ -0,0 +1,1193 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Multi-phase, multi-block election provider pallet
+//!
+//! This pallet manages the NPoS election across its different phases, with the ability to accept
+//! both on-chain and off-chain solutions. The off-chain solutions may be submitted as a signed or
+//! unsigned transaction. Crucially, it supports paginated, multi-block elections. The goal of
+//! supporting paged elections is to scale the elections linearly with the number of blocks
+//! allocated to the election.
+//!
+//! **PoV-friendly**: The work performed in a block by this pallet must be measurable and the time
+//! and computation required per block must be deterministic with regards to the number of voters,
+//! targets and any other configurations that are set apriori. These assumptions make this pallet
+//! suitable to run on a parachain.
+//!
+//! ## Pallet organization and sub-pallets
+//!
+//! This pallet consists of a `core` pallet and multiple sub-pallets. Conceptually, the pallets have
+//! a hierarquical relationship, where the `core` pallet sets and manages shared state between all
+//! pallets. Its "child" pallets can not depend on the `core` pallet and iteract with it through
+//! trait interfaces:
+//!
+//!```text
+//! 	pallet-core
+//! 		|
+//! 		|
+//! 		- pallet-verifier
+//! 		|
+//! 		- pallet-signed
+//! 		|
+//! 		- pallet-unsigned
+//! ```
+//!
+//! Each sub-pallet has a specific set of tasks and implement one or more interfaces to expose their
+//! functionality to the core pallet:
+//! - The [`verifier`] pallet provides an implementation of the [`verifier::Verifier`] trait, which
+//!   exposes the functionality to verify NPoS solutions in a multi-block context. In addition, it
+//!   implements [`verifier::AsyncVerifier`] which verifies multi-paged solution asynchronously.
+//! - The [`signed`] pallet implements the signed phase, where off-chain entities commit to and
+//!   submit their election solutions. This pallet implements the
+//!   [`verifier::SolutionDataProvider`], which is used by the [`verifier`] pallet to fetch solution
+//!   data to perform the solution verification.
+//! - The [`unsigned`] pallet implements the unsigned phase, where block authors can compute and
+//!   submit through inherent paged solutions. This pallet also implements the
```

Comment:
My understanding, based on https://github.com/paritytech/polkadot-sdk/blob/gpestana/pallet-epm-mb/substrate/frame/election-provider-multi-phase/src/lib.rs#L98, is that this phase is essentially as the signed one but the origins of solutions are different, is it not the case?
This description is a little unclear as-well, thus my remark here.